src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useRef, useState } from "react";
+import ArrowDownIcon from "../icons/ArrowDownIcon";
+import styles from "./Dropdown.module.scss";
+
+export type Option = {
+  key: string;
+  value: string;
+};
+
+export type DropdownProps = {
+  /** Массив возможных вариантов для выбора */
+  options: Option[];
+  /** Текущий индекс, может быть пустым */
+  keySelected: string;
+  /** Callback, вызываемый при выборе варианта */
+  onChange: (key: string) => void;
+};
+
+const Dropdown: React.FC<DropdownProps> = ({
+  options,
+  keySelected,
+  onChange,
+}: DropdownProps) => {
+  let [isExpanded, Toggle] = useState(false);
+  let [index, SetIndex] = useState(keySelected);
+
+  const newRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    document.addEventListener("mousedown", handleOutsideClick);
+    return () => {
+      document.removeEventListener("mousedown", handleOutsideClick);
+    };
+  });
```

Comment:
```suggestion
  }, []);
```
Так думаю, нужно навесить обработчик при монтировании. Без массива зависимостей этот эффект будет выззываться при каждом перерендере